File: softwareDev/tempConversion/temp.py
#temp conversion program
def celsius_to_fahrenheit(celsius):
    return (celsius * 9/5) + 32
def fahrenheit_to_celsius(fahrenheit):
    return (fahrenheit - 32) * 5/9
def kelvin_to_celsius(kelvin):
    return kelvin - 273.15
def celsius_to_kelvin(celsius):
    return celsius + 273.15
def kelvin_to_fahrenheit(kelvin):
    return (kelvin - 273.15) * 9/5 + 32
def fahrenheit_to_kelvin(fahrenheit):
    return (fahrenheit - 32) * 5/9 + 273.15
def main():
    print("Temperature Conversion Program")
    print("1. Celsius to Fahrenheit")
    print("2. Fahrenheit to Celsius")
    print("3. Kelvin to Celsius")
    print("4. Celsius to Kelvin")
    print("5. Kelvin to Fahrenheit")
    print("6. Fahrenheit to Kelvin")
    choice = input("Choose a conversion (1-6): ")
    temp = float(input("Enter the temperature to convert: "))
    
    if choice == '1':
        result = celsius_to_fahrenheit(temp)
        print(f"{temp}°C is {result}°F")
    elif choice == '2':
        result = fahrenheit_to_celsius(temp)
        print(f"{temp}°F is {result}°C")
    elif choice == '3':
        result = kelvin_to_celsius(temp)
        print(f"{temp}K is {result}°C")
    elif choice == '4':
        result = celsius_to_kelvin(temp)
        print(f"{temp}°C is {result}K")
    elif choice == '5':
        result = kelvin_to_fahrenheit(temp)
        print(f"{temp}K is {result}°F")
    elif choice == '6':
        result = fahrenheit_to_kelvin(temp)
        print(f"{temp}°F is {result}K")
    else:
        print("Invalid choice")
if __name__ == "__main__":
    main()

File: softwareDev/tempConversion/temp.cpp
#include <iostream>
#include <cmath>

using namespace std;

double celsiusToFahrenheit(double celsius) {
    return (celsius * 9.0 / 5.0) + 32.0;
}
double fahrenheitToCelsius(double fahrenheit) {
    return (fahrenheit - 32.0) * 5.0 / 9.0;
}
double celsiusToKelvin(double celsius) {
    return celsius + 273.15;
}
double kelvinToCelsius(double kelvin) {
    return kelvin - 273.15;
}
double fahrenheitToKelvin(double fahrenheit) {
    return celsiusToKelvin(fahrenheitToCelsius(fahrenheit));
}
double kelvinToFahrenheit(double kelvin) {
    return celsiusToFahrenheit(kelvinToCelsius(kelvin));
}
void displayMenu() {
    cout << "Temperature Conversion Menu:" << endl;
    cout << "1. Celsius to Fahrenheit" << endl;
    cout << "2. Fahrenheit to Celsius" << endl;
    cout << "3. Celsius to Kelvin" << endl;
    cout << "4. Kelvin to Celsius" << endl;
    cout << "5. Fahrenheit to Kelvin" << endl;
    cout << "6. Kelvin to Fahrenheit" << endl;
    cout << "7. Exit" << endl;
    cout << "Choose an option (1-7): ";
}
int main() {
    int choice;
    double temperature, convertedTemp;

    do {
        displayMenu();
        cin >> choice;

        if (choice >= 1 && choice <= 6) {
            cout << "Enter temperature: ";
            cin >> temperature;
        }

        switch (choice) {
            case 1:
                convertedTemp = celsiusToFahrenheit(temperature);
                cout << temperature << " °C = " << convertedTemp << " °F" << endl;
                break;
            case 2:
                convertedTemp = fahrenheitToCelsius(temperature);
                cout << temperature << " °F = " << convertedTemp << " °C" << endl;
                break;
            case 3:
                convertedTemp = celsiusToKelvin(temperature);
                cout << temperature << " °C = " << convertedTemp << " K" << endl;
                break;
            case 4:
                convertedTemp = kelvinToCelsius(temperature);
                cout << temperature << " K = " << convertedTemp << " °C" << endl;
                break;
            case 5:
                convertedTemp = fahrenheitToKelvin(temperature);
                cout << temperature << " °F = " << convertedTemp << " K" << endl;
                break;
            case 6:
                convertedTemp = kelvinToFahrenheit(temperature);
                cout << temperature << " K = " << convertedTemp << " °F" << endl;
                break;
            case 7:
                cout << "Exiting the program." << endl;
                break;
            default:
                cout << "Invalid choice. Please try again." << endl;
        }
        cout << endl;
    } while (choice != 7);

    return 0;
}